test/vm/unit/runtime/storage_tests.cpp

```diff
@@ -49,352 +49,150 @@ TEST_F(RuntimeTest, TransientStorage)
-
-    // current == original
-    auto &loc =
-        host_.accounts[ctx_.env.recipient].storage[bytes32_from_uint256(key)];
-    loc.original = bytes32_from_uint256(val);
-    loc.current = bytes32_from_uint256(val);
-
-    ctx_.gas_remaining = 0;
-    ASSERT_EQ(load(key), val);
-
-    // nonempty -> same nonempty
-    ctx_.gas_remaining = 2301;
-    store(key, val);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(load(key), val);
-
-    // nonempty -> different nonempty
-    ctx_.gas_remaining = 4200;
-    store(key, val_2);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-    ASSERT_EQ(load(key), val_2);
-
-    // nonempty -> empty
-    ctx_.gas_remaining = 2301;
-    store(key, 0);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(ctx_.gas_refund, 15000);
-    ASSERT_EQ(load(key), 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinLoadCold)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-
-    ctx_.gas_remaining = 2000;
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinLoadWarm)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-
-    host_.access_storage(ctx_.env.recipient, bytes32_from_uint256(key));
-
-    ctx_.gas_remaining = 0;
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinOriginalEmpty)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-    auto store = wrap(sstore<traits>);
-
-    // empty -> nonempty (cold)
-    ctx_.gas_remaining = 22000;
-    store(key, val);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-    ASSERT_EQ(load(key), val);
-
-    // nonempty -> nonempty (warm)
-    ctx_.gas_remaining = 2301;
-    store(key, val_2);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(load(key), val_2);
-
-    // nonempty -> empty (warm)
-    ctx_.gas_remaining = 2301;
-    store(key, 0);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(ctx_.gas_refund, 19900);
-    ASSERT_EQ(load(key), 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinOriginalNonEmpty)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-    auto store = wrap(sstore<traits>);
+TYPED_TEST(RuntimeTraitsTest, StorageLoadCold)
+{
+    using traits = TestFixture::Trait;
+    auto load = TestFixture::wrap(sload<traits>);
+
+    this->ctx_.gas_remaining = [] {
+        if constexpr (is_monad_trait_v<traits>) {
+            if constexpr (traits::monad_rev() >= MONAD_SEVEN) {
+                return 8000;
+            }
+        }
+        if constexpr (traits::evm_rev() <= EVMC_ISTANBUL) {
+            return 0;
+        }
+        else {
+            return 2000;
+        }
+    }();
+    ASSERT_EQ(load(key), 0);
+    ASSERT_EQ(this->ctx_.gas_remaining, 0);
+    ASSERT_EQ(load(key), 0);
+    ASSERT_EQ(this->ctx_.gas_remaining, 0);
```

Comment:
Want to assert status code here? I suppose the first `load` is expected to be success and the second `load` will fail with out-of-gas.